[android] Can not perform this action after onSaveInstanceState

A new regression after merging #4174:

```
java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState

     FATAL EXCEPTION: Thread-2
Process: app.organicmaps.debug, PID: 9950
java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
	at androidx.fragment.app.FragmentManager.checkStateLoss(FragmentManager.java:1610)
	at androidx.fragment.app.FragmentManager.ensureExecReady(FragmentManager.java:1711)
	at androidx.fragment.app.FragmentManager.execSingleAction(FragmentManager.java:1725)
	at androidx.fragment.app.BackStackRecord.commitNow(BackStackRecord.java:317)
	at app.organicmaps.widget.placepage.PlacePageController.removePlacePageFragments(PlacePageController.java:397)
	at app.organicmaps.widget.placepage.PlacePageController.onHiddenInternal(PlacePageController.java:183)
	at app.organicmaps.widget.placepage.PlacePageController.-$$Nest$monHiddenInternal(Unknown Source:0)
	at app.organicmaps.widget.placepage.PlacePageController$1.onSheetHidden(PlacePageController.java:100)
	at app.organicmaps.widget.placepage.DefaultBottomSheetCallback.onStateChanged(DefaultBottomSheetCallback.java:33)
	at com.google.android.material.bottomsheet.BottomSheetBehavior.setStateInternal(BottomSheetBehavior.java:1388)
	at com.google.android.material.bottomsheet.BottomSheetBehavior$StateSettlingTracker$1.run(BottomSheetBehavior.java:1984)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
	at android.view.Choreographer.doCallbacks(Choreographer.java:761)
	at android.view.Choreographer.doFrame(Choreographer.java:693)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at androidx.test.espresso.base.Interrogator.loopAndInterrogate(Interrogator.java:10)
	at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:7)
	at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:1)
	at androidx.test.espresso.base.UiControllerImpl.loopMainThreadForAtLeast(UiControllerImpl.java:7)
	at androidx.test.espresso.action.Tap$1.sendTap(Tap.java:4)
	at androidx.test.espresso.action.GeneralClickAction.perform(GeneralClickAction.java:4)
	at androidx.test.espresso.ViewInteraction$SingleExecutionViewAction.perform(ViewInteraction.java:2)
	at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:23)
	at androidx.test.espresso.ViewInteraction.-$$Nest$mdoPerform(Unknown Source:0)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:6)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:1)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6669)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

MediumPhone (ARM), Virtual, API level 28

https://console.firebase.google.com/u/0/project/omapsapp/testlab/histories/bh.7a38003b1edbc7b5/matrices/5196180644321985195/executions/bs.dca7c31e2afabb3a/issues

https://user-images.githubusercontent.com/170263/220216433-e9e26d55-3d8b-4f2b-8ef9-87ec2e10037d.mp4

The relevant log just before the crash:
```
I/ActivityManager(3304): Displayed app.organicmaps.debug/app.organicmaps.bookmarks.BookmarkCategoriesActivity: +157ms
02-19 21:24:53.739: I/GoogleInputMethod(7441): onFinishInput() : Dummy InputConnection bound
02-19 21:24:53.739: I/GoogleInputMethod(7441): onStartInput() : Dummy InputConnection bound
02-19 21:24:53.740: D/EGL_emulation(9950): eglMakeCurrent: 0x7104e4ca40: ver 3 0 (tinfo 0x7104fd1e60)
02-19 21:24:53.758: D/EGL_emulation(9950): eglMakeCurrent: 0x7104e4ca40: ver 3 0 (tinfo 0x7104fd1e60)
02-19 21:24:53.760: D/MapFragment(9950): MapFragment.java:68 surfaceDestroyed(): surfaceDestroyed
02-19 21:24:53.760: D/Map(9950): Map.java:188 onSurfaceDestroyed(): mSurfaceCreated = true, mSurfaceAttached = true, isAdded = true
02-19 21:24:53.760: I/OMcore(9950): organicmaps/Framework.cpp:285 DetachSurface(): Detach surface started. destroySurface = 1
02-19 21:24:53.760: I/OMcore(9950): organicmaps/Framework.cpp:298 DetachSurface(): Destroy surface.
02-19 21:24:53.784: I/OMcore(9950): drape_frontend/frontend_renderer.cpp:2276 OnContextDestroy(): On context destroy.
02-19 21:24:53.803: I/OMcore(9950): drape_frontend/backend_renderer.cpp:660 OnContextDestroy(): On context destroy.
02-19 21:24:53.826: D/OMcore(9950): drape_frontend/drape_engine.cpp:396 SetRenderingDisabled(): Rendering disabled
02-19 21:24:53.831: I/OMcore(9950): organicmaps/Framework.cpp:320 DetachSurface(): Detach surface finished.
02-19 21:24:53.836: D/EGL_emulation(9950): eglMakeCurrent: 0x7104e4ca40: ver 3 0 (tinfo 0x7104fd1e60)
02-19 21:24:53.840: D/AppBackgroundTracker(9950): AppBackgroundTracker.java:77 onActivityStopped(): onActivityStopped activity = app.organicmaps.MwmActivity@3ffae33
02-19 21:24:53.841: D/MapFragment(9950): MapFragment.java:112 onStop(): onStop
 I/BaseActivityDelegate(9950): BaseActivityDelegate.java:97 logLifecycleMethod(): MwmActivity: onStop() activity: app.organicmaps.MwmActivity@3ffae33
02-19 21:24:53.841: I/OMcore(9950): organicmaps/Framework.cpp:903 Java_app_organicmaps_Framework_nativeRemovePlacePageActivationListener(): Remove global map object listener
02-19 21:24:53.841: D/LocationHelper(9950): LocationHelper.java:420 removeListener(): listener: app.organicmaps.MwmActivity@3ffae33 count was: 2
02-19 21:24:53.841: D/LocationHelper(9950): LocationHelper.java:594 detach(): activity = app.organicmaps.MwmActivity@3ffae33
02-19 21:24:53.841: D/LifecycleMonitor(9950): Lifecycle status change: app.organicmaps.MwmActivity@3ffae33 in: STOPPED
02-19 21:24:53.842: D/AppBackgroundTracker(9950): AppBackgroundTracker.java:109 onActivitySaveInstanceState(): onActivitySaveInstanceState activity = app.organicmaps.MwmActivity@3ffae33
02-19 21:24:53.842: D/DefaultBottomSheetCallback(9950): DefaultBottomSheetCallback.java:25 onStateChanged(): State change, new = HIDDEN
```

I've attached the video. @Jean-BaptisteC maybe you can reproduce it...

Also I got another crash that may be relevant (but I can't explain how to repeat it):
```
02-21 01:18:44.077 31049 31049 I BaseActivityDelegate: MwmActivity: onStart() activity: app.organicmaps.MwmActivity@986726f
02-21 01:18:44.077 31049 31049 I OMcore  : organicmaps/Framework.cpp:863 Java_app_organicmaps_Framework_nativePlacePageActivationListener(): Set global map object listener
02-21 01:18:44.080 31049 31049 I BaseActivityDelegate: MwmActivity: onPostCreate() activity: app.organicmaps.MwmActivity@986726f
02-21 01:18:44.081 31049 31049 I BaseActivityDelegate: MwmActivity: onResume() activity: app.organicmaps.MwmActivity@986726f
02-21 01:18:44.095 31049 31049 I BaseActivityDelegate: MwmActivity: onPostResume() activity: app.organicmaps.MwmActivity@986726f
02-21 01:18:44.118 31049 31049 D AndroidRuntime: Shutting down VM
--------- beginning of crash
02-21 01:18:44.128 31049 31049 E AndroidRuntime: FATAL EXCEPTION: main
02-21 01:18:44.128 31049 31049 E AndroidRuntime: Process: app.organicmaps.beta, PID: 31049
02-21 01:18:44.128 31049 31049 E AndroidRuntime: java.lang.RuntimeException: Unable to destroy activity {app.organicmaps.beta/app.organicmaps.MwmActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'androidx.lifecycle.MutableLiveData app.organicmaps.widget.placepage.PlacePageViewModel.getMapObject()' on a null object reference
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4704)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4722)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:39)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1986)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:215)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6939)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:870)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'androidx.lifecycle.MutableLiveData app.organicmaps.widget.placepage.PlacePageViewModel.getMapObject()' on a null object reference
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at app.organicmaps.widget.placepage.PlacePageView.onDestroy(Unknown Source:8)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.Fragment.performDestroy(Unknown Source:19)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentStateManager.destroy(Unknown Source:147)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(Unknown Source:266)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentStore.moveToExpectedState(Unknown Source:30)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager.moveToState(Unknown Source:27)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager.dispatchStateChange(Unknown Source:9)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager.dispatchDestroy(Unknown Source:13)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentController.dispatchDestroy(Unknown Source:4)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentActivity.onDestroy(Unknown Source:5)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.appcompat.app.AppCompatActivity.onDestroy(Unknown Source:0)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at app.organicmaps.base.BaseMwmFragmentActivity.onDestroy(Unknown Source:0)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.Activity.performDestroy(Activity.java:7407)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1328)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4689)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	... 11 more
```

--- a/android/src/app/organicmaps/widget/placepage/PlacePageController.java
+++ b/android/src/app/organicmaps/widget/placepage/PlacePageController.java
@@ -84,7 +84,8 @@ public class PlacePageController implements Initializable<Activity>,
     public void onStateChanged(@NonNull View bottomSheet, int newState)
     {
       Logger.d(TAG, "State change, new = " + PlacePageUtils.toString(newState));
-      if (PlacePageUtils.isSettlingState(newState) || PlacePageUtils.isDraggingState(newState))
+      if (PlacePageUtils.isSettlingState(newState) || PlacePageUtils.isDraggingState(newState) ||
+          mMwmActivity.isFinishing() || mMwmActivity.isDestroyed())
         return;
 
       PlacePageUtils.moveViewportUp(mPlacePage, mViewportMinHeight);
@@ -96,6 +97,8 @@ public class PlacePageController implements Initializable<Activity>,
     @Override
     public void onSlide(@NonNull View bottomSheet, float slideOffset)
     {
+      if (mMwmActivity.isFinishing() || mMwmActivity.isDestroyed())
+        return;
       stopCustomPeekHeightAnimation();
       mDistanceToTop = bottomSheet.getTop();
       mSlideListener.onPlacePageSlide(mDistanceToTop);

